Lib9c/Action/ClaimMonsterCollectionReward.cs

```diff
@@ -41,54 +41,52 @@ public override IAccountStateDelta Execute(IActionContext context)
                 throw new FailedLoadStateException($"Aborted as the monster collection state failed to load.");
             }
 
-            MonsterCollectionState monsterCollectionState = new MonsterCollectionState(stateDict);
+            var monsterCollectionState = new MonsterCollectionState(stateDict);
 
-            if (!monsterCollectionState.CanReceive(context.BlockIndex))
+            int step = (int)Math.DivRem(
+                context.BlockIndex - monsterCollectionState.StartedBlockIndex,
+                MonsterCollectionState.RewardInterval,
+                out _
+            );
+            if (monsterCollectionState.ReceivedBlockIndex > 0)
             {
-                throw new RequiredBlockIndexException(
-                    $"{collectionAddress} is not available yet; it will be available after {Math.Max(monsterCollectionState.StartedBlockIndex, monsterCollectionState.ReceivedBlockIndex) + MonsterCollectionState.RewardInterval}");
+                int previousStep = (int)Math.DivRem(
+                    monsterCollectionState.ReceivedBlockIndex - monsterCollectionState.StartedBlockIndex,
+                    MonsterCollectionState.RewardInterval,
+                    out _
+                );
+                step -= previousStep;
```

Comment:
IMO, it would be better to create method for this logic. because, It looks like it will be used in the NineChronicles.Headless API too.